src/pkg/cluster/zarf.go

```diff
@@ -74,6 +74,65 @@ func (c *Cluster) GetDeployedPackage(ctx context.Context, packageName string) (*
 	return deployedPackage, nil
 }
 
+// UpdateDeployedPackage updates the deployed package metadata.
+func (c *Cluster) UpdateDeployedPackage(ctx context.Context, depPkg types.DeployedPackage) error {
+	secretName := config.ZarfPackagePrefix + depPkg.Name
+	packageSecretData, err := json.Marshal(depPkg)
+	if err != nil {
+		return err
+	}
+	packageSecret := &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: corev1.SchemeGroupVersion.String(),
+			Kind:       "Secret",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: ZarfNamespaceName,
+			Labels: map[string]string{
+				ZarfManagedByLabel:   "zarf",
+				ZarfPackageInfoLabel: depPkg.Name,
+			},
+		},
+		Type: corev1.SecretTypeOpaque,
+		Data: map[string][]byte{
+			"data": packageSecretData,
+		},
+	}
+	err = func() error {
+		_, err := c.Clientset.CoreV1().Secrets(packageSecret.Namespace).Get(ctx, packageSecret.Name, metav1.GetOptions{})
+		if err != nil && !kerrors.IsNotFound(err) {
+			return err
+		}
+		if kerrors.IsNotFound(err) {
+			_, err = c.Clientset.CoreV1().Secrets(packageSecret.Namespace).Create(ctx, packageSecret, metav1.CreateOptions{})
+			if err != nil {
+				return fmt.Errorf("unable to create the zarf state secret: %w", err)
```

Comment:
This secret isn't Zarf state, but the existing remove also uses this text, so up to you if you'd like to change it here or in a separate PR.

Reply:
No reason not to fix this now.